TUTORIAL.md

```diff
@@ -239,6 +241,52 @@ give_largest_value(a: 1, b: 2, c: 3) # returns 3
 give_largest_value("a" => 1, 2 => 2, c: 3)
 ```
 
+### Contracts On Keyword Arguments
+
+ruby 2.0+, but can be used for normal hashes too, when keyword arguments are
+not available
+
+Lets say you are writing a simple function and require a bunch of keyword arguments:
+
+```ruby
+def connect(host, port:, user:, password:)
+```
+
+You can of course put `Hash` contract on it:
+
+```ruby
+Contract String, { :port => Num, :user => String, :password => String } => Connection
+def connect(host, port:, user:, password:)
+```
+
+But this will not quite work if you want to have a default values:
+
+```ruby
+Contract String, { :port => Num, :user => String, :password => String } => Connection
+def connect(host, port: 5000, user:, password:)
+```
+
+```
+ContractError: Contract violation for argument 2 of 2:
```

Comment:
Just before this, can you add a line that calls the `connect` function that will throw the related ContractError, and make it obvious that we are not passing in a value for `port`?